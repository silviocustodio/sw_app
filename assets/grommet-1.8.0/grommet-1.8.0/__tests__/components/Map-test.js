// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Map from '../../src/js/components/Map';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('Map', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Map vertical={false} data={{
        "categories": [
          {
            "id": "category-1",
            "label": "First category",
            "items": [
              {"id": "item-1-1", "node": "First item"},
              {"id": "item-1-2", "node": "Second item"},
              {"id": "item-1-3", "node": "Third item"}
            ]
          },
          {
            "id": "category-2",
            "label": "Second category",
            "items": [
              {"id": "item-2-1", "node": "Fourth item"},
              {"id": "item-2-2", "node": "Fifth item"}
            ]
          },
          {
            "id": "category-3",
            "label": "Third category",
            "items": [
              {"id": "item-3-1", "node": "Sixth item"},
              {"id": "item-3-2", "node": "Seventh item"}
            ]
          }
        ],
        "links": [
          {"parentId": "item-1-1", "childId": "item-2-2"},
          {"parentId": "item-1-2", "childId": "item-2-2"},
          {"parentId": "item-1-2", "childId": "item-2-1"},
          {"parentId": "item-2-2", "childId": "item-3-1"},
          {"parentId": "item-2-1", "childId": "item-3-2"}
        ]
      }} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
