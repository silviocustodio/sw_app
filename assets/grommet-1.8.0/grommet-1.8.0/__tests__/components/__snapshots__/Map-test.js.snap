// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map has correct default options 1`] = `
<div
  className="grommetux-map"
>
  <svg
    className="grommetux-map__links"
    height={100}
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 100 100"
    width={100}
  />
  <ol
    className="grommetux-map__categories"
  >
    <li
      className="grommetux-map__category"
    >
      <div
        className="grommetux-map__category-label"
      >
        First category
      </div>
      <ul
        className="grommetux-map__category-items"
      >
        <li
          aria-label="First item, Children: (Fifth item)"
          className="grommetux-map__item"
          id="item-1-1"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          First item
        </li>
        <li
          aria-label="Second item, Children: (Fifth item,Fourth item)"
          className="grommetux-map__item"
          id="item-1-2"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          Second item
        </li>
        <li
          aria-label="Third item, No Relationship"
          className="grommetux-map__item"
          id="item-1-3"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          Third item
        </li>
      </ul>
    </li>
    <li
      className="grommetux-map__category"
    >
      <div
        className="grommetux-map__category-label"
      >
        Second category
      </div>
      <ul
        className="grommetux-map__category-items"
      >
        <li
          aria-label="Fourth item, Parents: (Second item), Children: (Seventh item)"
          className="grommetux-map__item"
          id="item-2-1"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          Fourth item
        </li>
        <li
          aria-label="Fifth item, Parents: (First item,Second item), Children: (Sixth item)"
          className="grommetux-map__item"
          id="item-2-2"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          Fifth item
        </li>
      </ul>
    </li>
    <li
      className="grommetux-map__category"
    >
      <div
        className="grommetux-map__category-label"
      >
        Third category
      </div>
      <ul
        className="grommetux-map__category-items"
      >
        <li
          aria-label="Sixth item, Parents: (Fifth item)"
          className="grommetux-map__item"
          id="item-3-1"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          Sixth item
        </li>
        <li
          aria-label="Seventh item, Parents: (Fourth item)"
          className="grommetux-map__item"
          id="item-3-2"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          Seventh item
        </li>
      </ul>
    </li>
  </ol>
</div>
`;
